server/index.js

```diff
@@ -51,6 +51,91 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const sql = `
+      select "c"."cartItemId",
+             "c"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+       where "c"."cartId" = $1
+    `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => res.json(result.rows))
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (productId === undefined) {
+    return next(new ClientError('productId must be a positive integer'), 400);
+  }
+  const sql = `
+    select "price"
+      from "products"
+      where "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(price => {
+      const productPrice = price.rows[0];
+      if (!productPrice) {
+        return next(new ClientError(`Cannot find product with Id ${productId}`, 400));
```

Comment:
You should `throw` or use `Promise.reject()` here to shortcut to the `catch()` callback. As written, it's possible for your promise chain to continue and cause an internal server error.